<?xml version="1.0"?>
<!-- vim: set fenc=utf-8 et sts=4 ft=xml : -->
<launch>

    <node name="filter"
        ns="/processes/odometry"
        pkg="pose_ukf"
        type="pose_ukf_node"
        output="screen"
        >
        <!--
        launch-prefix="tmux_new_window gdb - -args "
        launch-prefix="tmux_new_window gdb -ex run - -args "
        launch-prefix="tmux_new_window valgrind "
        -->
        <remap from="imu" to="/io/microstrain_3dm_gx3_35/imu"/>
        <remap from="joint_state" to="/motion/platform_joint_state"/>
        <param name="wheel_diameter" type="double" value="0.13"/>
        <param name="wheel_forward_noise" type="double" value="0.001"/>
        <param name="wheel_perpendicular_noise" type="double" value="0.0005"/>
        <rosparam param="wheels">
            -
                steering_joint_name: port_steering_joint
                axle_joint_name: port_axle_joint
                wheel_link: port_wheel
            -
                steering_joint_name: starboard_steering_joint
                axle_joint_name: starboard_axle_joint
                wheel_link: starboard_wheel
            -
                steering_joint_name: stern_steering_joint
                axle_joint_name: stern_axle_joint
                wheel_link: stern_wheel
        </rosparam>

        <rosparam param="process_sigma">
            position: [ 0.001, 0.001 ]
            velocity: [ 0.5, 0.5 ]
            acceleration: [ 1.5, 1.5 ]
            orientation: [ 0.001, 0.001, 0.01 ]
            omega: [ 0.01, 0.01, 0.01 ]
            gyro_bias: [ 1.0e-12, 1.0e-12, 1.0e-12 ]
            accel_bias: [ 1.0e-12, 1.0e-12, 1.0e-12 ]
        </rosparam>

        <!-- position, velocity, acceleration, orientation(r,p,y), omega, gyro bias, accel bias -->
        <rosparam param="initial_covariance">
            [ 1.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 1.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 1.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 1.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 1.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 1.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.01, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.01, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.10, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.01, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.01, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.01, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 1e-12, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 1e-12, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 1e-12, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 1e-12, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 1e-12, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 1e-12 ]
        </rosparam>
        <!-- position, velocity, orientation(r,p,y), omega, gyro bias, accel bias -->
        <!--
        <rosparam param="process_noise">
            [ 0.01, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.01, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.001, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.10, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.10, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.05, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.10, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.10, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.20, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.05, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.05, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.10, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.001, 0.000, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.001, 0.000, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.001, 0.000, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.001, 0.000, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.001, 0.000,
              0.00, 0.00, 0.000, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0.000, 0.000, 0.000, 0.000, 0.000, 0.001 ]
        </rosparam>
        -->
    </node>

</launch>
